.gitbook/contents/developer/features/aa-basics/hybrid-compute.md

```diff
@@ -0,0 +1,549 @@
+
+## Overview
+
+```mermaid
+graph LR
+A[Bundler] -- method --> B[Offchain Handler] -- update --> D[Smart Contract]
+```
+
+### About Hybrid Compute
+
+These are some basic information that might give you an easier time understanding this repository.
+Hybrid Compute in the Boba Network allows smart contracts to interact with external data and services. Typically, smart
+contracts on blockchains like Ethereum are limited to the data available on the blockchain, unable to access outside
+information directly. Hybrid Compute changes this by enabling smart contracts to make API calls to external services.
+This interaction allows smart contracts to access off-chain data and perform complex computations off-chain. The results
+of these computations can then be used on-chain, enhancing the functionality and efficiency of smart contracts. By doing
+so, Hybrid Compute reduces gas costs associated with complex computations and broadens the scope of what decentralized
+applications (dApps) can achieve. In essence, it bridges the gap between the blockchain and the real world, allowing for
+more sophisticated and dynamic applications.
+
+### About a Bundler
+
+An Account Abstraction Bundler is a component of a blockchain system designed to enhance the functionality and
+flexibility of smart contract accounts. It facilitates the direct control of accounts by translating user intents, removing the
+traditional division between externally owned accounts and their dependencies for contract accounts. This abstraction allows users to
+interact more seamlessly and manage smart contract accounts directly. Additionally, the bundler aspect optimizes
+transaction packaging by grouping multiple transactions together before adding them to a block. This optimization can
+lead to more efficient use of block space and potentially lower transaction fees. Overall, Account Abstraction Bundlers
+enable the development of more sophisticated decentralized applications (dApps) by empowering smart contracts with
+greater control and functionality within the blockchain network
+
+## Prerequisites
+
+Before you begin, make sure you have the following prerequisites in place:
+
+- [ ] API key: Obtain an API key from our system by reaching out on [Discord](https://discord.com/invite/Hvu3zpFwWd).
+- [ ] Development environment: Set up your development environment with the necessary tools and dependencies. In this
+  tutorial, make sure to have to following tools and technologies set up
+  - Python
+  - Docker
+
+## Step 1: Writing the offchain handler
+
+The first step is to write our ``offchain handler.`` This handler is responsible for receiving requests from the bundler
+along with their payloads and returning the appropriate responses.
+
+Let's begin with a simple example where the handler receives two numbers. It will perform both addition and subtraction
+on these numbers. If the result of the subtraction results in an underflow (i.e., the first number is greater than the
```

Comment:
Underflow is when the second number is greater than the first (reversed from what's written here).